components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImplTest.java

```diff
@@ -0,0 +1,233 @@
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import org.slf4j.Logger;
+
+public class DefaultPatientsResourceImplTest {
+
+    @Rule
+    public MockitoComponentMockingRule<PatientsResource> mocker =
+        new MockitoComponentMockingRule<PatientsResource>(DefaultPatientsResourceImpl.class);
+
+    @Mock
+    private User currentUser;
+
+    @Mock
+    private Logger logger;
+
+    private PatientRepository repository;
+
+    private QueryManager queries;
+
+    private AuthorizationManager access;
+
+    private UserManager users;
+
+    private EntityReferenceResolver<EntityReference> currentResolver;
+
+    private DomainObjectFactory factory;
+
+    private DocumentReference userProfileDocument;
+
+    private DocumentReference patientDocument;
+
+    @Mock
+    private Patient patient;
+
+    private URI uri;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    private DefaultPatientsResourceImpl patientsResource;
+
+    private XWikiContext context;
+
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException {
+        MockitoAnnotations.initMocks(this);
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+        ComponentManager compManager = this.mocker.getInstance(ComponentManager.class, "context");
+        Provider<XWikiContext> provider = this.mocker.getInstance(XWikiContext.TYPE_PROVIDER);
+        this.context = provider.get();
+        when(compManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(this.context).when(executionContext).getProperty("xwikicontext");
+
+        this.repository = this.mocker.getInstance(PatientRepository.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.patientsResource = (DefaultPatientsResourceImpl)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.queries = this.mocker.getInstance(QueryManager.class);
+        this.uri = new URI("http://uri");
+        this.userProfileDocument = new DocumentReference("wiki", "user", "00000001");
+
+        doReturn(this.uri).when(this.uriInfo).getBaseUri();
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+        ReflectionUtils.setFieldValue(this.patientsResource, "uriInfo", this.uriInfo);
+
+        doReturn("P00000001").when(this.patient).getId();
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(this.userProfileDocument).when(this.currentUser).getProfileDocument();
+    }
+
+    @Test
+    public void addPatientUserDoesNotHaveAccess() throws XWikiRestException {
```

Comment:
`throws XWikiRestException` has been removed from the Class/Interface you are testing, remove them here as well